ros/src/waypoint_updater/waypoint_updater.py

```diff
@@ -94,12 +96,15 @@ def find_next_waypoint(self):
             ego_pose = self.ego.pose
             n_waypoints = len(self.waypoints)
             # Find the closest waypoint
-            for i in range(n_waypoints):
-                wp_pos = self.waypoints[i].pose.pose.position
+            for i in range(self.next_idx, self.next_idx + n_waypoints):
+                idx = (i + n_waypoints) % n_waypoints
+                wp_pos = self.waypoints[idx].pose.pose.position
                 dl = self.euclidean(ego_pose.position, wp_pos)
                 if dl < min_dist:
                     min_dist = dl
-                    min_idx = i
+                    min_idx = idx
+                if min_dist < 10 and dl > min_dist:
```

Comment:
Will this value `10` also work in the test track? I'm wondering if it's specific to how waypoints are spaced in the simulator

Reply:
Oh, sure it is a very arbitrary number. The way to find the next waypoint will be changed in another pull request which will not rely on such value, however `10` is an OK number for now, since there's no way that the track is shorter than 10 meters as well as there will be at least a few waypoints for any 10 meters of the track.

Reply:
ok!